sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsModelFactory.cs

```diff
@@ -264,26 +264,13 @@ public static DetectLanguageResultCollection DetectLanguageResultCollection(IEnu
         /// <param name="text">Sets the <see cref="CategorizedEntity.Text"/> property.</param>
         /// <param name="category">Sets the <see cref="CategorizedEntity.Category"/> property.</param>
         /// <param name="subCategory">Sets the <see cref="CategorizedEntity.SubCategory"/> property.</param>
-        /// <param name="score">Sets the <see cref="CategorizedEntity.ConfidenceScore"/> property.</param>
-        /// <returns>A new instance of <see cref="TextAnalytics.CategorizedEntity"/> for mocking purposes.</returns>
-        [EditorBrowsable(EditorBrowsableState.Never)]
-        public static CategorizedEntity CategorizedEntity(string text, string category, string subCategory, double score)
-        {
-            return new CategorizedEntity(new Entity(text, category, subCategory, default, default, score));
-        }
-
-        /// <summary>
-        /// Initializes a new instance of <see cref="TextAnalytics.CategorizedEntity"/> for mocking purposes.
-        /// </summary>
-        /// <param name="text">Sets the <see cref="CategorizedEntity.Text"/> property.</param>
-        /// <param name="category">Sets the <see cref="CategorizedEntity.Category"/> property.</param>
-        /// <param name="subCategory">Sets the <see cref="CategorizedEntity.SubCategory"/> property.</param>
-        /// <param name="score">Sets the <see cref="CategorizedEntity.ConfidenceScore"/> property.</param>
         /// <param name="offset">Sets the <see cref="CategorizedEntity.Offset"/> property.</param>
+        /// <param name="length">Sets the <see cref="CategorizedEntity.Length"/> property.</param>
+        /// <param name="score">Sets the <see cref="CategorizedEntity.ConfidenceScore"/> property.</param>
         /// <returns>A new instance of <see cref="TextAnalytics.CategorizedEntity"/> for mocking purposes.</returns>
-        public static CategorizedEntity CategorizedEntity(string text, string category, string subCategory, double score, int offset)
+        public static CategorizedEntity CategorizedEntity(string text, string category, string subCategory, int offset, int length, double score)
```

Comment:
nit: consider leaving `double score, int offset, int length` for order so that if a user was using the previous method (that doesn't have offset and length), they can just add the two properties instead of changing parameters